snapshot/generate.go

```diff
@@ -615,31 +616,18 @@ func (dl *diskLayer) generate(stats *generatorStats) {
 			snapAccountWriteCounter.Inc(time.Since(start).Nanoseconds())
 			return nil
 		}
-		// Retrieve the current account and flatten it into the internal format
-		var acc struct {
-			Nonce    uint64
-			Balance  *big.Int
-			Root     common.Hash
-			CodeHash []byte
-		}
-		if err := rlp.DecodeBytes(val, &acc); err != nil {
+		serializer := account.NewAccountSerializer()
+		if err := rlp.DecodeBytes(val, serializer); err != nil {
 			logger.Crit("Invalid account encountered during snapshot creation", "err", err)
 		}
+		acc := serializer.GetAccount()
 		// If the account is not yet in-progress, write it out
 		if accMarker == nil || !bytes.Equal(accountHash[:], accMarker) {
 			dataLen := len(val) // Approximate size, saves us a round of RLP-encoding
 			if !write {
-				if bytes.Equal(acc.CodeHash, emptyCode[:]) {
-					dataLen -= 32
-				}
-				if acc.Root == emptyRoot {
-					dataLen -= 32
-				}
```

Comment:
Why did you remove this logic? Is it unnecessary in Klaytn?

Reply:
@hqjang-pepper This is for approximate data size. So, Etheruem uses slim account in order to save the space. It reduces `dataLen` since `nil` is stored instead of `emptyRootHash` or `emptyCodeHash`.